src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/termfrequency/TermCleaningRepository.kt

```diff
@@ -0,0 +1,92 @@
+package uk.gov.justice.digital.hmpps.personrecord.jpa.repository.termfrequency
+
+import jakarta.persistence.EntityManager
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+
+@Component
+class TermCleaningRepository(
+  private val entityManager: EntityManager,
+) {
+
+  @Transactional
+  fun deletePersonAggregateData() = entityManager.createNativeQuery(DELETE_PERSON_AGGREGATE_SQL).executeUpdate()
+
+  @Transactional
+  fun collectPersonAggregateData() = entityManager.createNativeQuery(COLLECT_PERSON_AGGREGATE_SQL).executeUpdate()
+
+  companion object {
+    private const val COLLECT_PERSON_AGGREGATE_SQL = """
+      CREATE MATERIALIZED VIEW personrecordservice.person_aggregate_data as (
```

Comment:
is this a good idea?
What about anything which depends on the view running while this job is running?
Should we have this defined in flyway instead and refresh it?
Maybe one for the 11am meeting?

Reply:
The impression I got from the meeting today was that it would be better to refresh the view rather than drop and recreate it. This would also give us the advantage of the view appearing in a database schema report (do we do one for this schema 🤔 we should do). It would also be better to have the CREATE VIEW code in Flyway so we can find it easily